src/server/event_bus.c

```
@@ -0,0 +1,219 @@
+    if (!event_name || !fn) return -1;
+    if (priority < 0)   priority = 0;
+    if (priority > 255) priority = 255;
+
+    obc_event_entry_t *e = find_or_create_entry(event_name);
+    if (!e) return -1;
+
+    if (e->fire_depth > 0) {
+        /* Defer addition: we're currently iterating subs for this event.
+         * The new handler will be inserted after the fire cycle completes
+         * and will NOT run in the same obc_event_fire() invocation. */
+        if (e->sub_count + e->add_count >= OBC_EVENT_MAX_SUBS) return -1;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)